src/pages/index.js

```diff
@@ -88,15 +182,56 @@ variables.profileEditButtonSelector.addEventListener("click", () => {
   openEditForm();
 });
 
-variables.profileAddButtonSelector.addEventListener("click", () =>
-  openAddForm()
+// POPUP WITH FORM - AVATAR FORM
+
+const avatarModal = new PopupWithForm(
+  variables.avatarModal,
+  handleAvatarFormSubmit
 );
+avatarModal.setEventListeners();
+
+function openAvatarForm() {
+  avatarModal.open();
+}
+
+variables.avatarIconSelector.addEventListener("click", () => {
+  openAvatarForm();
+});
+
+function handleAvatarFormSubmit(inputValues) {
+  setButtonText(variables.avatarModalButton, "Saving...");
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```